### DISCUSSION TITLE
On Updating Projects Page

### DISCUSSION THREAD

**Discussion by astroesteban** [nasa/fprime#844] [General]
Hey y'all! I noticed the [projects page](https://nasa.github.io/fprime/projects.html) is blank despite there being [blog posts from JPL](https://www.jpl.nasa.gov/news/meet-the-open-source-software-powering-nasas-ingenuity-mars-helicopter) about some projects using the framework. It's even missing the very famous Ingenuity on there.

**I was wondering if that page could be updated to list some missions that are using the F' framework?** 

I know it's impossible to know about every single project using F' but a good start would be the ones with a direct relationship to JPL like  Ingenuity, ASTERIA, ISS-RapidScat, Lunar Flashlight, and NEAScout. I think having something on that page could add an extra "wow" factor to this already incredible project.

**Comment by ThibFrgsGmz** [comment]
Personally, I think it is a good idea and we could go further by making a table with all the projects and programs implementing fprime.
In this table we would find :

| | | | | | | | |
|:---|:---|:---|:---|:---|:---|:---|:---| 
|**_Project/Program_**| **_SW Class (A to F)_** | **_NASA Center / Company_** | **_Cost estimate_** | **_OS_** | **_HW Platform_** | **_Launch Date_** | **_Note_**
| Ingenuity | ? | NASA/JPL | ? | Linux | Qualcomm’s Snapdragon 801 | July 30, 2020  | ? |

Of course, if some information is not made public, we could put a "?" sign.

The table I describe is the same as the one in the report "Open Source Core Flight System (cFS) Flight Software (FSW) Verification & Validation (V&V) Final Summary: IV&V Analysis Technical Report" page 38 available at https://ntrs.nasa.gov/citations/20205010026.

**Reply by astroesteban** [reply]
That's a good format and one that is already used by other NASA projects which is a big plus.

**Reply by LeStarch** [reply]
I'll see if I can get some projects to respond with data to fill in a similar table.  Maybe some images too?

**Reply by astroesteban** [reply]
If it's not too much work that would be awesome!

**Reply by ThibFrgsGmz** [reply]
I don't know if integrating the images directly into the table would be viable in the sense that some projects might not have images and that would break the homogeneity of the table (at least in the representation I have of the table). On the other hand, we could reference a link to the project presentation page.

If some of these projects are open source we could also reference their URLs to see how F Prime has been implemented in them.

I think that this type of table could reassure and encourage companies to use F Prime by allowing them to compare their project to a project that has already been developed with F Prime and find similarities.

**Reply by astroesteban** [reply]
I agree with your points @ThibFrgsGmz. I think going with the format you provided above will be the best approach. It provides a quick overview of some of the projects using F' plus some (optional) links for additional context or information users can reference.

**Reply by ThibFrgsGmz** [reply]
Maybe the project.md file (https://github.com/nasa/fprime/blob/devel/docs/projects.md) shall be updated accordingly, or deleted due to duplication/maintenance or it could redirect to the corresponding page on [F Prime github.io](https://nasa.github.io/fprime/projects.html).

**Reply by astroesteban** [reply]
If I'm not mistaken it looks like the F' website is just GitHub Pages so it _should_ pull from the markdown files in this repo.

**Reply by ThibFrgsGmz** [reply]
Now that I think about it, with this table we assume that the project is necessarily flight-proven, i.e. at TRL 9.

Perhaps it would be interesting to add a column indicating the TRLs (Technology Readiness Levels) associated with the projects. Thus, we could include projects whose TRL level is lower, a minimum of level 6 for example, i.e. "Demonstration of a model or prototype of a system/subsystem in an operational environment".

@LeStarch Did you manage to gather information to make this kind of table?

It would also be interesting to create an "Add projects to the F Prime projects page" thread and pin it to the Discussion tab. This would help gain visibility so people would think about referencing their projects if they want/can. What do you think about this?

**Reply by LeStarch** [reply]
My apologies, I was unable to see past a certain number of contexts.  

1. Yes Github pages drives the F´ site, so updating docs/projects.md would take effect (after merging to devel).
2. Getting information is in-progress.
3. I don't mind having discussions to collect the following data:
  - Formal projects using F´
  - "Awesome" contributions  for an eventual awesome repository.  (We kind-of started this in that one discussion).
